segments/helpers.py

```diff
@@ -112,17 +103,25 @@ def refresh_segment(self, segment_id, sql):
             self.segment_member_refresh_key,
             new_key
         )
-        for user_key in (self.segment_member_key % i for i in self.redis.sscan_iter(new_key)):
-            self.redis.sadd(user_key, segment_id)
+        with self.redis.pipeline(transaction=False) as pipeline:
+            for user_id in self.redis.sscan_iter(new_key, count=REDIS_SSCAN_COUNT):
+                pipeline.sadd(self.segment_member_key % user_id, segment_id)
+                if len(pipeline) >= BATCH_SIZE:
+                    pipeline.execute()
+            pipeline.execute()
```

Comment:
Can we move this to a function call? It overlaps with line 119 for the most part other than the pipeline operation. I think you can basically push this and just send the operation to do to the function. Something like

```
run_pipeline(new_key, REDIS_SSCAN_COUNT, "sadd")

def run_pipeline(new_key, count, operation):
  with self.redis.pipeline(transaction=False) as pipeline:
      for user_id in self.redis.sscan_iter(new_key, count=REDIS_SSCAN_COUNT):
          pipeline.operation(self.segment_member_key % user_id, segment_id)
          if len(pipeline) >= BATCH_SIZE:
              pipeline.execute()
      pipeline.execute()
  

```